rclcpp/include/rclcpp/node.hpp

```diff
@@ -139,24 +139,84 @@ class Node : public std::enable_shared_from_this<Node>
   const std::vector<rclcpp::callback_group::CallbackGroup::WeakPtr> &
   get_callback_groups() const;
 
-  /// Create and return a Publisher.
+  /// Create and return a Publisher. Note: This constructor is deprecated
   /**
    * \param[in] topic_name The topic for this publisher to publish on.
    * \param[in] group The callback group for this publisher. NULL for no callback group.
    * \param[in] options Additional options to control creation of the publisher.
    * \return Shared pointer to the created publisher.
    */
+  template<
+    typename MessageT, typename Alloc = std::allocator<void>,
+    typename PublisherT = ::rclcpp::Publisher<MessageT, Alloc>>
+  std::shared_ptr<PublisherT>
+  create_publisher(
+    const std::string & topic_name,
+    size_t qos_history_depth,
+    std::shared_ptr<Alloc> allocator = nullptr);
+
+  /// Create and return a Publisher. Note: this constructor is deprecated
+  /**
+   * \param[in] topic_name The topic for this publisher to publish on.
+   * \param[in] qos_profile The quality of service profile to pass on to the rmw implementation.
+   * \param[in] allocator Optional custom allocator.
+   * \return Shared pointer to the created publisher.
+   */
+  template<
+    typename MessageT, typename Alloc = std::allocator<void>,
+    typename PublisherT = ::rclcpp::Publisher<MessageT, Alloc>>
+  std::shared_ptr<PublisherT>
+  create_publisher(
+    const std::string & topic_name,
+    const rmw_qos_profile_t & qos_profile = rmw_qos_profile_default,
+    std::shared_ptr<Alloc> allocator = nullptr);
+
+  /// Create and return a Publisher.
+  /**
+   * \param[in] topic_name The topic for this publisher to publish on.
+   * \param[in] options Additional options for the created Publisher
+   * \return Shared pointer to the created publisher.
+   */
   template<
     typename MessageT,
     typename Alloc = std::allocator<void>,
     typename PublisherT = ::rclcpp::Publisher<MessageT, Alloc>>
   std::shared_ptr<PublisherT>
   create_publisher(
     const std::string & topic_name,
+    const PublisherOptions<Alloc> & options,
```

Comment:
We decided to make options non-optional?

Reply:
Correct - otherwise this function is ambiguous with preexisting interfaces. Once the old interfaces have been removed, then we can give this argument a default value. However for now, it is expected that the function signature being called when you only specify a topic_name is the old function